File: delivery-service/delivery/api/src/main/java/org/example/delivery/api/domain/token/interfaces/TokenHelper.java
package org.example.delivery.api.domain.token.interfaces;

import org.example.delivery.api.domain.token.model.TokenDto;

import java.util.Map;

public interface TokenHelper {

    TokenDto issueAccessToken(Map<String, Object> data);

    TokenDto issueRefreshToken(Map<String, Object> data);

    Map<String, Object> validationTokenWithThrow(String token);
}

File: delivery/account/src/main/kotlin/org/example/delivery/account/domain/token/interfaces/TokenHelper.kt
package org.example.delivery.account.domain.token.interfaces

import org.example.delivery.account.domain.token.model.TokenDto

interface TokenHelper {

    fun issueAccessToken(data: Map<String, Any>): TokenDto?
    fun issueRefreshToken(data: Map<String, Any>): TokenDto?
    fun validationTokenWithThrow(token: String?): Map<String, Any>?
}